lineapy/api/api_classes.py

```diff
@@ -67,14 +70,27 @@ def get_value(self) -> object:
         """
         Get and return the value of the artifact
         """
-        value = self.db.get_node_value_path(self._node_id, self._execution_id)
-        if value is None:
+        pickle_filename = self.db.get_node_value_path(
+            self._node_id, self._execution_id
+        )
+        if pickle_filename is None:
             return None
         else:
             # TODO - set unicode etc here
             track(GetValueEvent(has_value=True))
-            with open(value, "rb") as f:
-                return FilePickler.load(f)
+            filepath = UPath(
+                options.safe_get("artifact_storage_dir")
+            ).joinpath(pickle_filename)
+            logger.debug(
+                f"Retriving pickle file from {filepath.as_posix()} ",
+            )
+            with get_handle(
+                filepath,
+                mode="rb",
+                is_text=False,
+                storage_options=options.storage_options,
+            ) as handles:
```

Comment:
as mentioned in the other comment, this will make it so that we do not save the backend info. so if there was a change in artifact_storage_dir, it might fail.